docker/openvm/Dockerfile

```diff
@@ -5,6 +5,26 @@ FROM ${BASE_IMAGE_TAG}
 # The ere-base image provides Rust, Cargo, and common tools.
 # We operate as root for SDK installation.
 
+# If current environment is in CI or not.
+ARG CI
+
+# Install CUDA Toolkit 12.9
+# If argument `CI` is set, we skip the installation.
+RUN [ -n "$CI" ] || \
+    (wget https://developer.download.nvidia.com/compute/cuda/repos/$(. /etc/os-release && echo "${ID}${VERSION_ID}" | tr -d '.')/$(uname -i)/cuda-keyring_1.1-1_all.deb && \
+    dpkg -i cuda-keyring_1.1-1_all.deb && \
+    rm cuda-keyring_1.1-1_all.deb && \
+    apt update && \
+    apt install -y cuda-toolkit-12-9 && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*)
+
+# Add nvcc to path
+ENV PATH=/usr/local/cuda/bin:$PATH
+
+# Default to build for RTX 50 series
+ARG CUDA_ARCH=120
+ENV CUDA_ARCH=${CUDA_ARCH}
```

Comment:
Uhm, I was wondering earlier what this `CUDA_ARCH` was for.

Do you know what happens if the host has a 4090 and doesn't set `CUDA_ARCH` to the right value? If the program crashes, then I think that might be okay but might still feel a bit cryptic.

I found https://crates.io/crates/nvml-wrapper/0.11.0 which could be interesting to auto-detect the right thing maybe? Long ago in workload, I did an even [more rawish](https://github.com/eth-act/zkevm-benchmark-workload/blob/0e60e902bed06580766f1f70fc889229ea75bc89/crates/metrics/src/lib.rs#L77) use of the `nvidia-smi`. 

If that could work, might be a more reliable way to avoid misconfigs and forget about this.